src/callback_handler/crypto.rs

```diff
@@ -274,15 +277,15 @@ iDAKBggqhkjOPQQDAgNIADBFAiEArSsdE5dDXqAU2vM3ThT8GvTnjkWhER3l9v1j
             cert_chain: Some(cert_chain),
             not_after: None,
         };
-        assert_eq!(
-            verify_certificate(req).unwrap_err().to_string(),
-            "Certificate not issued by a trusted root"
-        );
-        Ok(())
+        let _reason = "Certificate is not trusted by the provided cert chain".to_string();
```

Comment:
this here is weird; I can't instantiate `reason` inside of the `matches!()`, but if I do it outside as it is, it says it's unused.

I couldn't find a simpler way to compare the output to the enum that I want for the tests (besides making the enum implement PartialEq), so I prefer to do `_reason` here for simplicity, as these are tests.

Reply:
yeah, let's stick with this approach. But please rename `_reason` to something that doesn't start with an underscore. This symbol is reserved to identify variables that are not used, but this one is actually used.

Reply:
that's the thing, I do need the underscore, and it's being used (as proof by the tests passing by checking the correct strings). If I don't use the underscore, I will get warnings on all the tests.

Reply:
oh, that's super weird! Ignore my comment then. Maybe add a simple inline comment about that...